File: fibonacci.cpp
#include <iostream>

using std::cout;
using std::endl;

int fibonacci(int n){
	int memo[n];

	memo[0] = 0;
	memo[1] = 1;

	for (int i = 2; i < n; ++i){
		memo[i] = memo[i - 2] + memo[i - 1];
	}
	return memo[n - 1] + memo[n - 2];
}

int main(){

	cout<<fibonacci(15)<<endl;
	cout<<fibonacci(20)<<endl;

}

File: fibonacci.js


function fibonacci(num, result = [0, 1, 1]) {
	if (result[num]) {
		return result[num];
	}
	result[num] = fibonacci(num - 1, result) + fibonacci(num - 2, result);
	return result[num];
}

console.log(fibonacci(144));

